search-sentiments-map/public/scripts/modal-script.js

```diff
@@ -154,59 +185,81 @@ function highlightWords(
     snippet, wordsToHighlight, scoreIsPositive, resultElement) {
   let startDisplayingSnippetIndex = 0;
   wordsToHighlight.forEach((word) => {
+    // The sentiment node.js module always returns the positive or negative 
+    // words in lowercase, so make only the snippet lowercase.
     let wordIndex = snippet.toLowerCase().indexOf(word);
     if (wordIndex !== -1){
-      let positiveHighlight = '<span style="color: ' +
-          POSITIVE_COLOR + ';">' +
-          snippet.substring(wordIndex, wordIndex + word.length) +
-          '</span>';
-      let negativeHighlight = '<span style="color: ' +
-          NEGATIVE_COLOR + ';">' +
-          snippet.substring(wordIndex,  wordIndex + word.length) +
+      let highlightHTML = '<span style="color: ' +
+          (scoreIsPositive ? POSITIVE_COLOR : NEGATIVE_COLOR) +
+          ';">' + snippet.substring(wordIndex, wordIndex + word.length) +
           '</span>';
-      snippet = scoreIsPositive ?
-          snippet.replace(new RegExp( '\\b'+ word + '\\b', 'gi'), positiveHighlight) : 
-          snippet.replace(new RegExp('\\b' + word + '\\b', 'gi'), negativeHighlight);
+      snippet = snippet.replace(new RegExp('\\b' + word + '\\b', 'gi'), highlightHTML);
     }
   });
   resultElement.innerHTML += snippet;
 }
 
 /** 
- * Displays the popularity timeline for the current search topic in the 
+ * Sets the popularity timeline for the current search topic in the 
  * selected country's modal.
- * @param {string} countryCode The two-letter code of the selected country.
  */
-function displayPopularityTimeline(countryCode) {
+function setPopularityTimeline() {
   const popularityTimelineElement = document.getElementById('popularity-timeline-tab');
   popularityTimelineElement.innerHTML = '';
-  fetch('/trends/', {
+  postTrendsToGetPopularityTimelineData(getCurrentSearchData().topic, countryCode)
```

Comment:
I think you are calling a function with 2 params that only requires one??

Reply:
Also why do you have to pass getCurrentSearchData().topic instead of just getting it in the called function